File: general/src/main/java/com/workoutnow/general/dtos/TrainingFeedbackForm.java
package com.workoutnow.general.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Range;

import javax.validation.constraints.NotNull;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TrainingFeedbackForm {
    @NotNull
    private Long trainingId;
    @NotNull
    private Boolean wouldLikeToDoAgain;
    @Range(max = 10, min = 0)
    private Integer difficultyScale;
}

File: analytics/src/main/kotlin/com/workoutnow/analytics/dto/TrainingFeedbackForm.kt
package com.workoutnow.analytics.dto

import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.NotNull

data class TrainingFeedbackForm (
    @NotNull
    @JsonProperty("trainingId")
    val trainingId: String,
    @NotNull
    @JsonProperty("difficultyScale")
    val difficultyScale: Int,
    @NotNull
    @JsonProperty("wouldLikeToDoAgain")
    val wouldLikeToDoAgain: Boolean
)
